lib/msf/util/dot_net_deserialization/types/general.rb

```diff
@@ -50,7 +50,7 @@ def member_types
       binary_type_enums.each do |binary_type|
         additional_info = nil
         if has_additional_info?(binary_type)
-          additional_info = additional_infos[additional_info_index].value
+          additional_info = additional_infos[additional_info_index].snapshot
```

Comment:
The `ClassTypeInfo` does not have a `#value` method that returns a scalar, switch to using snapshot instead to fix this issue.